### PR TITLE
Throw error when jdbc connection cannot be created

### PR DESCRIPTION
**Pull request by kokosing**
Throw error when jdbc connection cannot be created

### CODE REVIEW

**Comment by kokosing** [comment]
I had a wrong Driver class in tempto-configuration.yml. Instead of failing fast, tempto raised NPE couple calls later.

File: tempto-core/src/main/java/io/prestosql/tempto/internal/query/NonPoolingJdbcDataSource.java [hunk 1]
```diff
@@ -46,7 +48,8 @@ public Connection getConnection()
             throws SQLException
     {
         Properties properties = prepareConnectionProperties();
-        return driver.connect(jdbcParamsState.url, properties);
+        Connection connection = driver.connect(jdbcParamsState.url, properties);
+        return requireNonNull(connection, "connection is null");
```
Comment (findepi) [inline tempto-core/src/main/java/io/prestosql/tempto/internal/query/NonPoolingJdbcDataSource.java]: Does it make sense to quote `jdbcParamsState.url`?
```suggestion
        return requireNonNull(connection, () -> format("connection is null for '%s'", jdbcParamsState.url));
```

File: tempto-core/src/main/java/io/prestosql/tempto/query/JdbcConnectionsPool.java [hunk 2]
```diff
@@ -35,6 +35,9 @@ public Connection connectionFor(JdbcConnectivityParamsState jdbcParamsState)
         }
 
         Connection connection = dataSources.get(jdbcParamsState).getConnection();
+        if (connection == null) {
+            throw new IllegalStateException("No connection was created for: " + jdbcParamsState.getName());
```
Comment (findepi) [inline tempto-core/src/main/java/io/prestosql/tempto/query/JdbcConnectionsPool.java:40]: maybe a comment:

```suggestion
            // this should never happen, `javax.sql.DataSource#getConnection()` should not return null
            throw new IllegalStateException("No connection was created for: " + jdbcParamsState.getName());
```

?

**Review by findepi — APPROVED** [review]
remember to squash

**Pull request merged**